File: cartipay-main/Pardakht.PardakhtPay/BackEnd/Pardakht.PardakhtPay.Shared.Models/Entities/UserSegmentGroup.cs
using System;
using Pardakht.PardakhtPay.Shared.Models.Models;

namespace Pardakht.PardakhtPay.Shared.Models.Entities
{
    public class UserSegmentGroup : BaseEntity, ITenantGuid, IOwnerGuid, IDeletedEntity
    {
        public string Name { get; set; }

        public bool IsActive { get; set; }

        public bool IsDeleted { get; set; }

        public bool IsDefault { get; set; }

        public bool IsMalicious { get; set; }

        public DateTime CreateDate { get; set; }

        public string TenantGuid { get; set; }

        public string OwnerGuid { get; set; }

        public int Order { get; set; }
    }
}

File: cartipay-main/Pardakht.PardakhtPay/FrontEnd/Pardakht.PardakhtPay.AdminPortal/src/app/models/user-segment-group.ts
import { GenericHelper } from "../helpers/generic";

export class UserSegmentGroup {

    id: number;

    name: string;

    isActive: boolean;

    isDeleted: boolean;

    isDefault: boolean;

    isMalicious: boolean;

    createDate: Date;

    tenantGuid: string;

    ownerGuid: string;

    order: number;

    items: UserSegment[] = [];

    constructor(data?: any) {
        GenericHelper.populateData(this, data);
    }
}

export class UserSegment {

    id: number;

    userSegmentGroupId: number;

    userSegmentCompareTypeId: number;

    userSegmentTypeId: number;

    value: string;
}

export class RegisteredPhoneNumbers {
    phoneNumber: string;
}